### ISSUE TITLE
Enhancement: Downloader Place Files Into Subfolder Named Per Book Title and/or Subtitle

### ISSUE THREAD

**Issue by Budlyte** [advplyr/audiobookshelf#34]
Currently the downloads go into the Audiobooks folder in my phone, which I guess is just a funny coincidence that I already had it there to manually download to with X-plore. But most (if not all) audiobook players need the files to be separated into subfolders, otherwise the player app will shove them all together into one "book".

Can the downloader in the mobile app please create a subfolder with the book name and/or subtitle for the files to be placed into?

**Comment by Budlyte** [comment]
Maybe this was also part of my old version, but I see it is now placing the files inside of a subfolder.  it looks like this subfolder is named per the internal server ID of the book, so that the app can stay in sync with the server once connected.

Is there possibly a way to have the app rename the folder after download, per the Title, and link the ID to the new path?

**Comment by advplyr** [comment]
It is not currently storing them in a subfolder, I'm not sure how yours are ending up in one. All files are just put in `/Audiobooks`.
I've been struggling with this.. Android storage is not fun to work with.

In my tests I am able to put them inside a folder, but I can't delete that folder afterwards. So when deleting the downloaded audiobook it is leaving an empty folder in `/Audiobooks`. I'm going to push a release that leaves empty folders until I can figure that out.

The filename will need to remain the internal audiobook id for now, otherwise it won't match properly.

**Comment by Budlyte** [comment]
yeah, I was mistaken. My file manager showed a folder-type image for the mp3 file and I was tricked.

Thanks for the update!

**Comment by advplyr** [comment]
`v0.8.3-beta` places them in directories, but just to re-iterate, the directory does not get deleted when deleting the download in the app.
Before the Android app makes it out of beta that will need to be resolved, as well the folder and filenames should be the audiobook title instead of the internal id as you suggested.
